c2h/generators.cu

```diff
@@ -527,20 +545,33 @@ INSTANTIATE(__nv_fp8_e4m3);
 #endif // _CCCL_HAS_NVFP8()
 INSTANTIATE(float);
 INSTANTIATE(double);
+INSTANTIATE(cuda::std::complex<float>);
+INSTANTIATE(cuda::std::complex<double>);
```

Comment:
Remark: `generators.cu` is already a pain to compile. I think we should add new specializations carefully and consider splitting the file into multiple translation units.